src/phoenix/datasets/validation.py

```diff
@@ -168,6 +200,34 @@ def _check_missing_columns(dataframe: DataFrame, schema: Schema) -> List[err.Val
             ):
                 missing_columns.append(emb_col_names.link_to_data_column_name)
 
+    if schema.prompt_column_names is not None:
+        if schema.prompt_column_names.vector_column_name not in existing_columns:
+            missing_columns.append(schema.prompt_column_names.vector_column_name)
+        if (
+            schema.prompt_column_names.raw_data_column_name is not None
+            and schema.prompt_column_names.raw_data_column_name not in existing_columns
+        ):
+            missing_columns.append(schema.prompt_column_names.raw_data_column_name)
+        if (
+            schema.prompt_column_names.link_to_data_column_name is not None
+            and schema.prompt_column_names.link_to_data_column_name not in existing_columns
+        ):
+            missing_columns.append(schema.prompt_column_names.link_to_data_column_name)
+
+    if schema.response_column_names is not None:
+        if schema.response_column_names.vector_column_name not in existing_columns:
+            missing_columns.append(schema.response_column_names.vector_column_name)
+        if (
+            schema.response_column_names.raw_data_column_name is not None
+            and schema.response_column_names.raw_data_column_name not in existing_columns
+        ):
+            missing_columns.append(schema.response_column_names.raw_data_column_name)
+        if (
+            schema.response_column_names.link_to_data_column_name is not None
+            and schema.response_column_names.link_to_data_column_name not in existing_columns
+        ):
+            missing_columns.append(schema.response_column_names.link_to_data_column_name)
```

Comment:
nit: this is 100% duplicate code except that the key is different. I would DRY this up since they are identical

Reply:
Done